agr_literature_service/api/crud/topic_entity_tag_crud.py

```diff
@@ -313,12 +314,32 @@ def validate_tags(db: Session, new_tag_obj: TopicEntityTagModel):
         else:
             validate_tags_already_in_db_with_negative_tag(new_tag_obj, related_tags_in_db)
     # Validate current tag with existing ones
-    related_validating_tags_in_db = [related_tag for related_tag in related_tags_in_db if
-                                     related_tag.topic_entity_tag_source.validation_type is not None]
-    validate_new_tag_with_existing_tags(new_tag_obj, related_validating_tags_in_db)
+    if validate_new_tag:
+        related_validating_tags_in_db = [related_tag for related_tag in related_tags_in_db if
+                                         related_tag.topic_entity_tag_source.validation_type is not None]
+        validate_new_tag_with_existing_tags(new_tag_obj, related_validating_tags_in_db)
     db.commit()
 
 
+def revalidate_all_tags(db: Session, email: str, delete_all_first: bool = False):
```

Comment:
I'm not sure if I should add more db.commit() in this function. What's currently implemented deletes all validation entries from the current tag first, then recalculate the new validation and adds the related entries, then commit (as part of the validate_tags function). If delete_all_first is selected then the first commit will delete all validation entries in the table and then add the first re-calculated entries. Does this sound good @azurebrd @sweng66 ?

Reply:
I'll defer to you both